src/Orleans/Configuration/ProviderConfiguration.cs

```diff
@@ -197,6 +197,7 @@ public class ProviderCategoryConfiguration
         public const string BOOTSTRAP_PROVIDER_CATEGORY_NAME = "Bootstrap";
         public const string STORAGE_PROVIDER_CATEGORY_NAME = "Storage";
         public const string STREAM_PROVIDER_CATEGORY_NAME = "Stream";
+        public const string STATISTICS_PROVIDER_CATEGORY_NAME = "StatisticsProviders";
```

Comment:
See `Silo.cs` line 473. I think the category name should be "Statistics". Maybe that's why `RegisterStatisticsProvider` is not functional? 

Or can you tell me which part of method `RegisterStatisticsProvider` method above is not functional? Maybe I can help a bit. And also I can help you push your build to run the functional tests and BVTs